crates/bevy_core_pipeline/src/oit/mod.rs

```diff
@@ -0,0 +1,272 @@
+//! # Order Independent Transparency (OIT)
+//!
+//! This implementation uses 2 passes.
+//!
+//! The first pass writes the depth and color of all the fragments to a big buffer.
+//! The buffer contains N layers for each pixel.
+//! This pass is essentially a forward pass.
+//!
+//! The second pass is a single fullscreen triangle pass that sorts all the fragments then blends them together
+//! and outputs the result to the screen.
+//!
+//! If you want to use OIT for your custom material you need to call `oit_draw(position, color)` in your fragment shader.
+//! You also need to make sure that your fragment shader doesn't output any colors.
+
+use bevy_app::prelude::*;
+use bevy_asset::{load_internal_asset, Handle};
+use bevy_ecs::prelude::*;
+use bevy_math::UVec2;
+use bevy_render::{
+    camera::{Camera, ExtractedCamera},
+    extract_component::{ExtractComponent, ExtractComponentPlugin},
+    render_graph::{RenderGraphApp, ViewNodeRunner},
+    render_resource::{BufferUsages, BufferVec, DynamicUniformBuffer, Shader, TextureUsages},
+    renderer::{RenderDevice, RenderQueue},
+    view::Msaa,
+    Render, RenderApp, RenderSet,
+};
+use bevy_utils::{tracing::trace, HashSet, Instant};
+use bevy_window::PrimaryWindow;
+use resolve::{
+    node::{OitResolveNode, OitResolvePass},
+    OitResolvePlugin,
+};
+
+use crate::core_3d::{
+    graph::{Core3d, Node3d},
+    Camera3d,
+};
+
+/// Module that defines the necesasry systems to resolve the OIT buffer and render it to the screen
+pub mod resolve;
+
+/// Shader handle for the shader that draws the transparent meshes to the OIT layers buffer
+pub const OIT_DRAW_SHADER_HANDLE: Handle<Shader> = Handle::weak_from_u128(4042527984320512);
+
+/// Used to identify which camera will use OIT to render transparent meshes
+/// and to configure OIT
+// TODO consider supporting multiple OIT techniques like WBOIT, Moment Based OIT,
+// depth peeling, stochastic transparency, ray tracing etc.
+// This should probably be done by adding an enum to this component
+#[derive(Component, Clone, Copy, ExtractComponent)]
+pub struct OrderIndependentTransparencySettings {
+    /// Controls how many layers will be used to compute the blending.
+    /// The more layers you use the more memory it will use but it will also give better results.
+    /// 8 is generally recommended, going above 16 is probably not worth it in the vast majority of cases
+    pub layer_count: u8,
+}
+
+impl Default for OrderIndependentTransparencySettings {
+    fn default() -> Self {
+        Self { layer_count: 8 }
+    }
+}
+
+/// Plugin needed to enable Order Independent Transparency
```

Comment:
Usually we put the most docs on the plugin, and then link to the camera settings struct. Module docs usually just give a 1-3 sentence description and then link to the plugin. Can you re-arrange docs to follow that pattern (unless you disagree)? I think meshlets do a good job of this structure.

Also missing a discussion on what platforms are supported by this plugin. Definitely not WebGL2 at least, but what about others? Are non-default limits or wgpu features needed?

Also I think it should include an explanation of why you would want OIT.